File: 169-MajorityElement/169-MajorityElement.js
// Last updated: 24/04/2026, 10:05:35
var majorityElement = function(nums) {
    var obj = {};
    var n = nums.length;

    for (var i = 0; i < n; i++) {
        obj[nums[i]] = (obj[nums[i]] || 0) + 1;

        if (obj[nums[i]] > n / 2) {
            return nums[i];
        }
    }
};

File: 169-MajorityElement/169-MajorityElement.cpp
// Last updated: 07/06/2026, 18:05:42
1class Solution {
2public:
3    int majorityElement(vector<int>& nums) {
4        unordered_map<int , int> ans;
5
6        for(int value:nums){
7            ans[value]++;
8        }
9
10        int freq = nums.size()/2;
11
12        for(auto it: ans){
13            if(it.second> freq){
14                return it.first;
15            }
16        }
17
18        return -1;
19    }
20};